diff --git a/src/Form/ParagraphEmbedDialog.php b/src/Form/ParagraphEmbedDialog.php
index 293a6aa..f53dd07 100644
--- a/src/Form/ParagraphEmbedDialog.php
+++ b/src/Form/ParagraphEmbedDialog.php
@@ -77,13 +77,22 @@ class ParagraphEmbedDialog extends EmbeddedParagraphsForm {
     $form['#prefix'] = '<div id="paragraphs-entity-embed-type-form-wrapper">';
     $form['#suffix'] = '</div>';

+    if (!empty($embed_button->type_settings['enable_single_use_paragraph'])) {
+      // Remove the label field on single use paragraph option.
+      $form['label']['widget'][0]['value']['#access'] = FALSE;
+      $default_label = $form['label']['widget'][0]['value']['#default_value'];
+      $form['label']['widget'][0]['value']['#default_value'] = !empty($default_label) ? $default_label : \Drupal::time()->getRequestTime();
+    }
+
     // On inserting new paragraph allow administrator to choose whether new
     // paragraph should be created or an existing one will be used.
     if (
       // Conditions: entity should be new.
       $this->entity->isNew() &&
       // Insert method should not been chosen yet.
-      $form_state->get('insert_method_choise') !== self::INSERT_METHOD_SELECTED
+      $form_state->get('insert_method_choise') !== self::INSERT_METHOD_SELECTED &&
+      // Don't ask to insert if single use paragraph setting is set.
+      empty($embed_button->type_settings['enable_single_use_paragraph'])
     ) {
       // Raise a flag that method needs to be selected.
       $form_state->set('insert_method_choise', self::INSERT_METHOD_PROMPTED);
diff --git a/src/Plugin/EmbedType/Paragraph.php b/src/Plugin/EmbedType/Paragraph.php
index ed236f3..b9b97fb 100644
--- a/src/Plugin/EmbedType/Paragraph.php
+++ b/src/Plugin/EmbedType/Paragraph.php
@@ -48,6 +48,12 @@ class Paragraph extends EmbedTypeBase implements ContainerFactoryPluginInterface
    */
   public function buildConfigurationForm(array $form, FormStateInterface $form_state) {

+    $form['enable_single_use_paragraph'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Only allow paragraphs to be used once'),
+      '#description' => $this->t('With this option checked, you can only use a paragraph once. This removes the option to insert an existing paragraph and also removes the need to label every paragraph which is created.'),
+      '#default_value' => $this->getConfigurationValue('enable_single_use_paragraph'),
+    ];
     $form['enable_paragraph_type_filter'] = [
       '#type' => 'checkbox',
       '#title' => $this->t('Filter which Paragraph types to be embed'),
